src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -11,66 +11,155 @@
+        } else {
+            ImmutableMap<String, String> errorData = ImmutableMap.of("errorMessage", "Your message contains content that may " + 
+                "be deemed offensive by others. Please revise your message and try again.");
+
+            final String out = getOutputString("error", errorData);
+
+            response.getWriter().println(out);
+        }
+    }
+
+    /**
+     * Returns the output string for the response. In other words,
+     * it sets up the soy template with the passed in data.
+     */
+    private String getOutputString(String templateName, ImmutableMap data) {
+        // File path starts in target/portfolio-1
+        SoyFileSet sfs = SoyFileSet
+            .builder()
+            .add(new File(ROOT_FILE_PATH + "src/main/java/templates/chat.soy"))
+            .build();
+        SoyTofu tofu = sfs.compileToTofu();
+
+        return tofu.newRenderer("templates.chat." + templateName).setData(data).render();
+    }
+
+    /**
+     * Iterates through the message query to put all the message text into a list.
+     * A map is then created the pass the list of messages into the template.
+     */
+    private ImmutableMap<String, ImmutableList<String>> getTemplateData(PreparedQuery preparedMessageQuery) {
+        // Creates list of the messages text
+        ImmutableList<String> messagesList = Streams.stream(preparedMessageQuery.asIterable()).map(message -> 
         (String) message.getProperty(MESSAGE_TEXT_PROPERTY)).collect(toImmutableList());
 
-    // Data will be passed in as a list of messages in a map (needed for template)
-    ImmutableMap<String, ImmutableList<String>> data = ImmutableMap.of("messages", messagesList);
-
-    // File path starts in target/portfolio-1
-    SoyFileSet sfs = SoyFileSet
-        .builder()
-        .add(new File("../../src/main/java/templates/chat.soy"))
-        .build();
-    SoyTofu tofu = sfs.compileToTofu();
-
-    final String out = tofu.newRenderer("templates.chat.chatPage").setData(data).render();
-    response.getWriter().println(out);
-  }
-
-/**
- * Is called by a button on the chat form. doPost takes the message and its
- * attributes (such as timestamp and user) and stores them into Datastore.
- */
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    final String messageText = (String) request.getParameter(MESSAGE_TEXT_PROPERTY);
-    final long timestamp = System.currentTimeMillis();
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity messageEntity = new Entity(MESSAGE_KIND);
-    messageEntity.setProperty(MESSAGE_TEXT_PROPERTY, messageText);
-    messageEntity.setProperty(TIMESTAMP_PROPERTY, timestamp);
-    datastore.put(messageEntity);
-
-    response.sendRedirect("/chat");
-  }
+        // Data will be passed in as a list of messages in a map (needed for template)
+        return ImmutableMap.of("messages", messagesList);
```

Comment:
Why is this being returned as a map? Will there ever be different keys other than "messages"?